### DISCUSSION TITLE
Is it really worth showing KFP Executor logs in the UI?

### DISCUSSION THREAD

**Discussion by HumairAK** [kubeflow/pipelines#10739] [General]
Given a simple pipeline, something like this: 

<details> 

<summary>hello.py</summary> 

```python
from kfp import dsl, compiler


@dsl.component(base_image="quay.io/opendatahub/ds-pipelines-ci-executor-image:v1.0")
def say_hello(name: str) -> str:
    hello_text = f'Hello, {name}!'
    print(hello_text)
    return hello_text


@dsl.pipeline
def hello_pipeline(recipient: str) -> str:
    hello_task = say_hello(name=recipient)
    return hello_task.output


if __name__ == '__main__':
    compiler.Compiler().compile(hello_pipeline, __file__ + '.yaml')
```

</details>

The log output in the UI looks like: 

![image](https://github.com/kubeflow/pipelines/assets/10904967/c65637a4-4ff0-425d-ac60-9cff16479ead)

From the perspective of the user I would think everything outside of the red square is irrelevant, ops related stuff that only a cluster/project ops admin should care about. From the perspective of the data scienctist, I would expect they just want to see their pipeline code stdout here. 

I would think we want to keep these logs in the executor pod (for ops persona), but not in the UI. I don't know how we make that happen since the UI is just direclty fetching pod logs, but it's worth discussing.

Posing this as a discussion, before considering an issue, to see others thoughts here.

**Comment by gmfrasca** [comment]
If the intent here is to remove sources of confusion, perhaps containing this (and other more technical metadata/components) within a 'Operations' or 'Power User' tab or sub-menu could help, without totally removing the functionality that may be useful to others

**Comment by milosjava** [comment]
yes, I think @gmfrasca has a point here. That way data scientists can focus on the actual output but also allowing others to use additional info to debug the issue

**Comment by chensun** [comment]
My two cents: logs are useful when something went wrong, there's a tradeoff of having that information available when needed vs. requiring extra steps to enable the log for troubleshooting.
That being said, from the screenshot shown above, there're two parts of systems logs, one from the KFP driver (line 1-17), the other from KFP SDK (executor) (line 19-36). I do feel like we may remove the driver's log completely. But the KFP SDK's log could be usable (common scenario is when there's pip installation error due to dependencies conflicts)

**Comment by rimolive** [comment]
Some other suggestions we could think of:
* Make the infra pods log messages as debug or trace level
* Add a dropdown to the Logs tab, and separate the logs by Launcher/Driver/Executor, having Executor(main task) as the default option.

**Comment by HumairAK** [comment]
Looks like we can agree on removing the driver logs, created a follow up issue here: https://github.com/kubeflow/pipelines/issues/11168
